src/sorting/quicksort-functional.js

```diff
@@ -0,0 +1,69 @@
+(function (exports) {
```

Comment:
Would you rename the file to `quicksort-declarative.js`? The current implementations are functional as well (use recursion and higher order functions).